src/Service.php

```diff
@@ -131,8 +131,10 @@ public function __construct(
      */
     public function getFieldDefinitionAdapter(ClassDefinition\Data $fieldDefinition, bool $considerInheritance)
     {
-        if ($this->filterLocator->has($fieldDefinition->fieldtype)) {
-            $adapter = $this->filterLocator->get($fieldDefinition->fieldtype);
+        $fieldType = ($fieldDefinition->fieldtype) ?: $fieldDefinition->getFieldType();
```

Comment:
shouldn't just go for 
```suggestion
        $fieldType = $fieldDefinition->getFieldType();
```
?

Reply:
Yes, that's also fine